### ISSUE TITLE
[Feature] Do not gate hooks timeout logs behind a `DEBUG` env

### ISSUE THREAD

**Issue by CyriacBr** [microsoft/playwright#23373]
I have a hanging test in CI with Playwright and I was only able to see what was wrong with a `DEBUG=*` env. Shouldn't Playwright normally log errors such as that without needing an increased log level?  

![image](https://github.com/microsoft/playwright/assets/38442110/6949c6b1-00a1-4c76-8c0a-f62d1b83069a)

**Comment by aslushnikov** [comment]
> Shouldn't Playwright normally log errors such as that without needing an increased log level?

@CyriacBr yes, these should be surfaced in the regular log. Any chance you can build a repro that illustrates this issue? We'd love to debug this locally to fix it.

**Comment by CyriacBr** [comment]
I solved the issue in my CI, it was a spawned child_process that was hanging the job it seemed, thought I am not 100% sure what happened. 

I've been trying to do a mini reproduction but to no avail.  Playwright indeed seems to immediately logs when a timeout occurs, regardless of if the test hangs after. I'm unsure how it's possible that I saw debug logs regarding this without an actual timeout log showing.  
I'll close for now and re-open if I manage to repro.
